lib/KDateRange/KDateCalendar.vue

```diff
@@ -157,8 +157,9 @@
         },
         numOfDays: 7,
         isFirstChoice: this.selectedStartDate == null ? true : false,
-        activeMonth: new Date().getMonth() - 1 == -1 ? 11 : new Date().getMonth() - 1,
-        activeYearStart: new Date().getFullYear(),
+        activeMonth:
```

Comment:
I'm curious to know why 1 is subtracted from the active month. It means that the activeMonth will always be the month before the lastAllowedDate right? But if we are in January, the previous month would cause the year to also have to be reduced, but I see that activeYearStart is taken only from getFullYear.

Reply:
This is because of how the Date() constructor in JavaScript sets the months. It uses an integer value to represent the month, beginning with 0 for January to 11 for December.

Reply:
Oh no, I meant what if the `lastAllowedDate` was for example `new Date(2024, 0, 1)`, then activeMonth would be 11, but activeYearStart would be 2024, so it would be December 2024 instead of December 2023. But I looked at the code and I see that this `activeYearStart` value is updated in the created hook, so this doesn't happen, so no problem.

Anyways I think we could avoid the ternary here, and the check in the created hook if we do something similar to get the previous month:
```javascript
    data() {
      const previousMonth = new Date(this.lastAllowedDate);
      previousMonth.setMonth(lastAllowedDate.getMonth() - 1);
      return {
        ...,
        activeMonth: previousMonth.getMonth(),
        activeYearStart: previousMonth.getFullYear(),
      };
    },
```
But it is totally optional, although I think it would clarify the intention that activeMonth and activeYearStart refer to the previous month of the lastAllowedDate.